finality-aleph/src/lib.rs

```diff
@@ -189,15 +189,16 @@ impl From<SpawnTaskHandle> for SpawnHandle {
     }
 }
 
-impl aleph_bft::SpawnHandle for SpawnHandle {
+impl SpawnHandle {
     fn spawn(&self, name: &'static str, task: impl Future<Output = ()> + Send + 'static) {
         self.0.spawn(name, task)
     }
+
     fn spawn_essential(
         &self,
         name: &'static str,
         task: impl Future<Output = ()> + Send + 'static,
-    ) -> TaskHandle {
+    ) -> Pin<Box<impl Future<Output = Result<(), ()>> + Send>> {
```

Comment:
Why this instead of `TaskHandle`?

Reply:
I didn't want to make the internal api of aleph-node to depend on aleph-bft. But actually all of this is a bit confusing. That spawn_essential in general is a bit confusing too - in sp_core the intention of analogous method is to take down the whole node in case of an error. In our case it is just a wrapper allowing to recognize that a task finished and it could be an external to method.

Reply:
We are nonetheless depend on alephBFT so I don't see any gain here.
As to the confusion, the problem is that we changed AlephBFT to meet audit requirements. That should be refactored now.